overlord/servicestate/quota_handlers.go

```diff
@@ -123,54 +124,110 @@ func (m *ServiceManager) doQuotaControl(t *state.Task, _ *tomb.Tomb) error {
 		opts := &ensureSnapServicesForGroupOptions{
 			allGrps: allGrps,
 		}
-		appsToRestartBySnap, err = ensureSnapServicesForGroup(st, t, grp, opts)
+		servicesAffected, err = ensureSnapServicesForGroup(st, t, grp, opts)
 		if err != nil {
 			return err
 		}
 
 		// All persistent modifications to disk are made and the
 		// modifications to state will be committed by the
-		// unlocking in restartSnapServices. If snapd gets
+		// unlocking at the end of this task. If snapd gets
 		// restarted before the end of this task, all the
 		// modifications would be redone, and those
 		// non-idempotent parts of the task would fail.
 		// For this reason we record together with the changes
 		// in state the fact that the changes were made,
 		// to avoid repeating them.
 		// What remains for this task handler is just to
-		// restart services which will happen regardless if we
-		// get rebooted after unlocking the state - if we got
-		// rebooted before unlocking the state, none of the
-		// changes we made to state would be persisted and we
-		// would run through everything above here again, but
-		// the second time around EnsureSnapServices would end
-		// up doing nothing since it is idempotent.  So in the
-		// rare case that snapd gets restarted but is not a
+		// refresh security profiles and restart services which
+		// will happen regardless if we get rebooted after
+		// unlocking the state - if we got rebooted before unlocking
+		// the state, none of the changes we made to state would
+		// be persisted and we would run through everything above
+		// here again, but the second time around EnsureSnapServices
+		// would end up doing nothing since it is idempotent.
+		// So in the rare case that snapd gets restarted but is not a
 		// reboot also record which services do need
 		// restarting. There is a small chance that services
 		// will be restarted again but is preferable to the
 		// quota not applying to them.
-		if err := rememberQuotaStateUpdated(t, appsToRestartBySnap); err != nil {
+		if err := rememberQuotaStateUpdated(t, servicesAffected, refreshProfiles); err != nil {
 			return err
 		}
-
 	}
 
-	if err := restartSnapServices(st, t, appsToRestartBySnap, perfTimings); err != nil {
```

Comment:
we now have only a place calling restartSnapServices,  which is ensureSnapServicesStateForGroup, maybe it should be merged into that, also that function merits a comment that it doesn't update profiles, I think given its current use case dealing with a snap fully going away it's fine, worth for you to double check if that reasoning is  correct? maybe given its single use it also merits a more precise name?

Reply:
I updated the doc string for `ensureSnapServicesStateForGroup` and `restartSnapServices` to clarify their usage so it hopefully will give an idea for anyone that views that code exactly what is happening in `ensureSnapServicesStateForGroup`, as you said it's used in combination with system removal of a snap so it should not cause a conflict currently. 

For a different name, I am taking suggestions as I had a hard time coming up with something more meaningful